generate_directional_sprites.py

```diff
@@ -0,0 +1,243 @@
+    #     print(f"Forcing resize of {output_file}")
+    #     subprocess.run([
+    #         "magick", output_file,
+    #         "-resize", f"{orig_width}x{orig_height}!",
+    #         output_file
+    #     ])
+
+# Main directories
+graphics_dir = "graphics"
+types = ["normal", "fast", "express", "turbo"]
+directions = ["north", "east", "south", "west"]
+
+# Create directories for each type and direction
+for type_name in types:
+    type_dir = os.path.join(graphics_dir, type_name)
+    ensure_dir(type_dir)
+    # Create direction subdirectories
+    for direction in directions:
+        dir_path = os.path.join(type_dir, direction)
+        ensure_dir(dir_path)
+
+# Original sprite files
+files = ["input.png", "output.png", "input-shadow.png", "output-shadow.png"]
+
+# First, process normal sprites for all directions
+normal_dir = os.path.join(graphics_dir, "normal")
+for file in files:
+    source_file = os.path.join(normal_dir, file)
+    
+    if not os.path.exists(source_file):
+        print(f"Error: {source_file} does not exist!")
+        continue
+    
+    # Get original dimensions for reference
+    orig_width, orig_height = get_image_dimensions(source_file)
+    print(f"Original dimensions for {file}: {orig_width}x{orig_height}")
+    
+    # WEST - No rotation (original direction)
+    west_file = os.path.join(normal_dir, "west", file)
+    print(f"Creating {west_file}")
+    subprocess.run(["magick", source_file, west_file])
+    
+    # EAST - Horizontal flip of west (mirrored)
+    east_file = os.path.join(normal_dir, "east", file)
+    print(f"Creating {east_file}")
+    subprocess.run([
+        "magick", source_file, 
+        "-flop", 
+        east_file
+    ])
+    
+    # Verify dimensions are unchanged
+    east_width, east_height = get_image_dimensions(east_file)
+    west_width, west_height = get_image_dimensions(west_file)
+    
+    # if east_width != orig_width or east_height != orig_height:
+    #     print(f"WARNING: East dimensions don't match original, fixing")
+    #     subprocess.run(["magick", east_file, "-resize", f"{orig_width}x{orig_height}!", east_file])
+    
+    # if west_width != orig_width or west_height != orig_height:
+    #     print(f"WARNING: West dimensions don't match original, fixing")
+    #     subprocess.run(["magick", west_file, "-resize", f"{orig_width}x{orig_height}!", west_file])
+        
+    # NORTH - Create from individual rotated frames
+    north_file = os.path.join(normal_dir, "north", file)
+    print(f"Creating {north_file}")
+    
+    # Create temp directories
+    west_frames_dir = f"temp_west_frames_{file}"
+    north_frames_dir = f"temp_north_frames_{file}"
+    ensure_dir(west_frames_dir)
+    ensure_dir(north_frames_dir)
+    
+    try:
+        # Extract frames from west file
+        extract_frames(west_file, west_frames_dir)
+        
+        # Rotate each frame 90 degrees and flip horizontally
```

Comment:
Nice, simple solution! I had planned to render each combination of direction and input/output with different cameras and scenes from Blender but that turned out to be a TON of work to setup and keep everything in sync😅

Reply:
I had no idea how do do anything in blender so I got creative haha